dapp/src/components/buySell/BuySellWidget.js

```diff
@@ -242,20 +242,30 @@ const BuySellWidget = ({
         vaultContract.autoAllocateThreshold(),
         vaultContract.rebaseThreshold(),
       ])
-
-      if (totalMintAmount.gte(allocateThreshold.mul(96).div(100))) {
-        // Define gas limit only when the amount is over threshold
-        gasLimit = gasLimits.MINT_ALLOCATE_GAS_LIMIT
-      } else if (totalMintAmount.gte(rebaseThreshold.mul(96).div(100))) {
-        gasLimit = gasLimits.MINT_REBASE_GAS_LIMIT
+      const involvingMultipleCoins = mintAddresses.length > 1
+      const aboveAllocateThreshold = totalMintAmount.gte(
+        allocateThreshold.mul(96).div(100)
+      )
+      const aboveRebaseThreshold = totalMintAmount.gte(
+        rebaseThreshold.mul(96).div(100)
+      )
+
```

Comment:
👇 I realize that nested conditionals aren't great, but this is much easier to reason about than the prior base + additions thing. If I'm correct that we don't need significantly more gas for the nth coin beyond 2, then this allows us to be more precise and be more declarative with the constants.